src/loci/formats/in/ZarrReader.java

```diff
@@ -833,8 +833,11 @@ public String[] getUsedFiles(boolean noPixels) {
     String zarrRootPath = currentId.substring(0, currentId.indexOf(".zarr") + 5);
     ArrayList<String> usedFiles = new ArrayList<String>();
     try (Stream<Path> paths = Files.walk(Paths.get(zarrRootPath))) {
-      paths.filter(Files::isRegularFile)
-              .forEach(path -> usedFiles.add(path.toFile().getAbsolutePath()));
+      paths.filter(Files::isRegularFile) 
+              .forEach(path -> {if (!noPixels || 
+                  (noPixels && (path.endsWith(".zgroup") || path.endsWith(".zattrs") || path.endsWith(".xml")))) 
```

Comment:
Note: for mkngff, the logic I used was "don't enter directories that don't have a .zgroup or .zarray". That might be safer if others start using the "add your own random file" strategy.